tests/unittests/RecommendationSystemTest.cpp

```diff
@@ -0,0 +1,667 @@
+    std::unordered_map<std::string, Function *> name2func;
+
+    for (auto *F : mod.getFunctions()) {
+      name2func[F->getName()] = F;
+    }
+
+    std::vector<DAGNode *> exeList;
+    int endPt = 0;
+    int curPt = 0;
+    // The first node is always the dummy node.
+    exeList.push_back(G);
+    endPt++;
+    while (curPt < endPt) {
+      DAGNode *dag = exeList.at(curPt);
+      // The root in a G is always a dummy function.
+      if (curPt > 0) {
+        ExecutionEngine EE{EE_.getBackend()->getBackendKind()};
+        Function *func = name2func[dag->name];
+        EE.compile(CompilationMode::Infer, func);
+        updateInputPlaceholders(bindings, vars, inputs);
+        EE.run(bindings);
+      }
+      for (int i = 0, e = dag->children.size(); i < e; i++) {
+        exeList.push_back(dag->children.at(i));
+        endPt++;
+      }
+      curPt++;
+    }
+  }
+
+  /// Create partitions to run and compare results.
+  void runPartitionedGraph(size_t numDevices, size_t memSize,
+                           Tensor referenceResult,
+                           PlaceholderBindings &bindings) {
+
+    assert(memSize > 0 && "Must set partitionerPerDeviceMemCapacity > 0.");
+    assert(numDevices > 0 && "Must set partitionerNumDevices > 0.");
+
+    std::cout << numDevices << " devices of size " << memSize << "\n";
+    std::vector<DeviceInfo> devices(numDevices, {memSize});
+    Partitioner myPartitioner(&mod_, devices);
+    EXIT_ON_ERR(myPartitioner.Partition());
+
+    DAGListTy myList = std::move(myPartitioner.getPartitionResult());
+    std::cout << "Partitions = " << mod_.getFunctions().size() << std::endl;
+    ASSERT_LE(mod_.getFunctions().size(), numDevices);
+    ASSERT_EQ(myList.size(), 1);
+    DAG &dag = myList.front();
+
+    // Run the paritioned graph and compare the results.
+    bindings.allocate(mod_.getPlaceholders());
+    bindings.allocate(mod_.getPlaceholders());
+    executeDAG(dag.root.get(), mod_, bindings, {}, {});
+
+    Tensor *resultTensor = bindings.get(result);
+    EXPECT_TRUE(referenceResult.isEqual(*resultTensor));
+  }
+};
+
+INSTANTIATE_TEST_CASE_P_FOR_BACKEND_TEST(RecSys, RecommendationSystemTest);
+
+/// Standard Tests
+/// These tests have three options:
+///   * quantizeSLWS  enables Int8 Fused Rowwise Quantization for the Sparse
+///     Embeddings (Int8 quantized values with float scale and offset).
+///   * quantizeFC    enables Int8 Fused Rowwise Quantization for FC weights and
+///     activations inside the MLPs.
+///   * convertToFP16 walks the graph at the end of constructing the graph and
+///     converts all FP32 nodes & tensors to FP16, meaning the graph will use
+///     FP16 for internal weights, biases and activations (when not already Int8
+///     quantized). Inputs and outputs are still FP32 but are immediately
+///     dropped to FP16 precision at the beginning of the graph.
+
+TEST_P(RecommendationSystemTest, RecSys_FP32) {
+  ENABLED_BACKENDS(CPU);
```

Comment:
Can you also enable tests on Habana (those that are passing now).

Reply:
this should be on Habana as well.

Reply:
It's literally the line below 😅

Reply:
Hm, ok, might miss it.